src/common/providers/__tests__/TranslatableProvider.test.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+
+import TranslatableProvider from '../TranslatableProvider';
+import TranslatableContext, {
+  initialContext,
+} from '../../contexts/TranslatableContext';
+import { CONTENT_LANGUAGES } from '../../constants';
+import LanguageTabs from '../../components/languageTab/LanguageTabs';
+import { Language } from '../../../domain/api/generatedTypes/graphql';
+
+// Mock LanguageTabs to avoid rendering it in tests
+vi.mock('../components/languageTab/LanguageTabs', () => {
+  return function MockLanguageTabs() {
+    return <div data-testid="mock-language-tabs">Mock Language Tabs</div>;
+  };
+});
+
+describe('TranslatableProvider', () => {
+  it('renders children within the provider', () => {
+    render(
+      <TranslatableProvider>
+        <div data-testid="child-element">Child Element</div>
+      </TranslatableProvider>
+    );
+
+    expect(screen.getByTestId('child-element')).toBeInTheDocument();
+  });
+
+  it('provides the correct context values', () => {
+    let contextValue: any;
+    render(
+      <TranslatableProvider>
+        <TranslatableContext.Consumer>
+          {(value) => {
+            contextValue = value;
+            return null;
+          }}
+        </TranslatableContext.Consumer>
+      </TranslatableProvider>
+    );
+
+    expect(contextValue).toBeDefined();
+    expect(contextValue.getSource).toBeInstanceOf(Function);
+    expect(contextValue.defaultLanguage).toBe(Language.Fi);
+    expect(contextValue.languages).toEqual(CONTENT_LANGUAGES);
+    expect(contextValue.selectLanguage).toBeInstanceOf(Function);
+    expect(contextValue.selectedLanguage).toBe(Language.Fi);
+    expect(contextValue.selector).toBeDefined();
+  });
+
+  it('uses the correct default values when not specified', () => {
+    let contextValue: any;
+    render(
+      <TranslatableProvider>
+        <TranslatableContext.Consumer>
+          {(value) => {
+            contextValue = value;
+            return null;
+          }}
+        </TranslatableContext.Consumer>
+      </TranslatableProvider>
+    );
+
+    expect(contextValue.defaultLanguage).toBe(Language.Fi);
+    expect(contextValue.languages).toEqual(CONTENT_LANGUAGES);
+  });
```

Comment:
Test case `provides the correct context values` tests everything that the test case `uses the correct default values when not specified` tests, and a bit more. Thus the latter one could be removed, or do you see value in keeping it as a separate test?